python/ray/rllib/agents/impala/vtrace_policy_graph.py

```diff
@@ -274,52 +283,33 @@ def make_time_major(tensor, drop_last=False):
-                vf_loss_coeff=self.config["vf_loss_coeff"],
-                entropy_coeff=self.config["entropy_coeff"],
-                clip_rho_threshold=self.config["vtrace_clip_rho_threshold"],
-                clip_pg_rho_threshold=self.config["vtrace_clip_pg_rho_threshold"])
-        else:
-            self.loss = MultiVTraceLoss(
-                actions=make_time_major(actions, True),
-                actions_logp=make_time_major(action_dist.logp(
-                    tf.unstack(actions, axis=1)), True),
-                actions_entropy=make_time_major(action_dist.entropy(), True),
-                dones=make_time_major(dones, True),
-                behaviour_logits=make_time_major(
-                    unpacked_behaviour_logits, True),
-                target_logits=make_time_major(unpacked_outputs, True),
-                discount=config["gamma"],
-                rewards=make_time_major(rewards, True),
-                values=make_time_major(values, True),
-                bootstrap_value=make_time_major(values)[-1],
-                valid_mask=make_time_major(mask, True),
-                vf_loss_coeff=self.config["vf_loss_coeff"],
-                entropy_coeff=self.config["entropy_coeff"],
-                clip_rho_threshold=self.config["vtrace_clip_rho_threshold"],
-                clip_pg_rho_threshold=self.config["vtrace_clip_pg_rho_threshold"])
-
-        # KL divergence between worker and learner logits for debugging
-        if self._is_discrete:
-            model_dist = Categorical(self.model.outputs)
-            behaviour_dist = Categorical(behaviour_logits)
-        else:
-            model_dist = MultiCategorical(unpacked_outputs)
-            behaviour_dist = MultiCategorical(unpacked_behaviour_logits)
+        self.loss = MultiVTraceLoss(
+            actions=make_time_major(loss_actions, drop_last=True),
+            actions_logp=make_time_major(action_dist.logp(actions),
+                                         drop_last=True),
+            actions_entropy=make_time_major(action_dist.entropy(),
+                                            drop_last=True),
+            dones=make_time_major(dones, drop_last=True),
+            behaviour_logits=make_time_major(
+                unpacked_behaviour_logits, drop_last=True),
+            target_logits=make_time_major(unpacked_outputs, drop_last=True),
+            discount=config["gamma"],
+            rewards=make_time_major(rewards, drop_last=True),
+            values=make_time_major(values, drop_last=True),
+            bootstrap_value=make_time_major(values)[-1],
+            valid_mask=make_time_major(mask, drop_last=True),
+            vf_loss_coeff=self.config["vf_loss_coeff"],
+            entropy_coeff=self.config["entropy_coeff"],
+            clip_rho_threshold=self.config["vtrace_clip_rho_threshold"],
+            clip_pg_rho_threshold=self.config["vtrace_clip_pg_rho_threshold"])
+
+        model_dist = dist_class(dist_inputs)
```

Comment:
This is not ok - in the vanilla ray they were Categorical even though dist_class was not (it was DiagGaussian). So revert to either Categorical or MultiCategorical, behind discrete if.